trl/trainer/online_dpo_trainer.py

```diff
@@ -428,18 +430,23 @@ def training_step(
             ref_logprobs = torch.take_along_dim(ref_all_logprobs, completion_ids.unsqueeze(-1), dim=2).squeeze(-1)
             del ref_output, ref_logits, ref_all_logprobs  # free memory
 
-        # Get the reward from the reward model or judge:
-        if self.judge is not None:
-            completions = self.processing_class.batch_decode(
-                prompt_completion_ids[:, context_length:], skip_special_tokens=True
-            )
-            completions = [completion.strip() for completion in completions]  # remove the leading space
+        # Decode the completions, and format them if the input is conversational
+        device = prompt_completion_ids.device
+        completions_ids = prompt_completion_ids[:, context_length:]
+        completions = self.processing_class.batch_decode(completions_ids, skip_special_tokens=True)
+        if is_conversational({"prompt": prompts[0]}):
+            completions = [[{"role": "assistant", "content": completion}] for completion in completions]
 
+        # Get the reward from the reward model or judge
+        if self.judge is not None:
+            # Once formatted, conversational data may contain special tokens (such as <|im_start|>) that are not
```

Comment:
I think one edge case here is when the prompt consists of multi-turn messages, but I think we can ignore this for now.

Reply:
Why would it be an edge-case?

Reply:
Ah I see: the simple chat template assigns alternating `User:` and `Assistant:` roles - ignore my comment!